pkg/storage/storage.go

```diff
@@ -62,84 +62,173 @@ func Remove(client *occlient.Client, name string, componentName string, applicat
-			}
-			log.Debugf("Removed storage for pvc: %v", pvc)
-		}
-	} else {
-		pvc := generatePVCNameFromStorageName(name)
-		if err := removeStorage(client, pvc, dc.Name); err != nil {
-			return errors.Wrap(err, "unable to remove storage")
-		}
+	pvcName, err := getPVCNameFromStorageName(client, name)
+	if err != nil {
+		return errors.Wrapf(err, "unable to get PVC for storage %v", name)
+	}
+
+	// delete the associated PVC with the component
+	err = client.DeletePVC(pvcName)
+	if err != nil {
+		return errors.Wrapf(err, "unable to delete PVC %v", pvcName)
 	}
 
 	return nil
 }
 
-// List lists all the storage associated with the given component of the given
+// List lists all the mounted storage associated with the given component of the given
 // application
 func List(client *occlient.Client, componentName string, applicationName string) ([]StorageInfo, error) {
-	labels := storagelabels.GetLabels("", componentName, applicationName, false)
-	selector := util.ConvertLabelsToSelector(labels)
+	componentLabels := componentlabels.GetLabels(componentName, applicationName, false)
+	componentSelector := util.ConvertLabelsToSelector(componentLabels)
 
-	log.Debugf("Looking for PVCs with the selector: %v", selector)
-	pvcs, err := client.GetPVCsFromSelector(selector)
+	dc, err := client.GetOneDeploymentConfigFromSelector(componentSelector)
 	if err != nil {
-		return nil, errors.Wrap(err, "unable to get PVC names")
+		return nil, errors.Wrapf(err, "unable to get Deployment Config associated with component %v", componentName)
+	}
+
+	mountedStorageMap := make(map[string]string)
+	volumeMounts := client.GetVolumeMountsFromDC(dc)
+	for _, volumeMount := range volumeMounts {
+		pvcName := client.GetPVCNameFromVolumeMountName(volumeMount.Name, dc)
+		if pvcName == "" {
+			return nil, fmt.Errorf("no PVC associated with Volume Mount %v", volumeMount.Name)
+		}
+		pvc, err := client.GetPVCFromName(pvcName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to get PVC %v", pvcName)
+		}
+
+		storageName := getStorageFromPVC(pvc)
+		mountedStorageMap[storageName] = volumeMount.MountPath
 	}
 
+	pvcs, err := client.GetPVCsFromSelector(storagelabels.StorageLabel)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Deployment Config associated with component %v", componentName)
+	}
 	var storageList []StorageInfo
 	for _, pvc := range pvcs {
-		storage := getStorageFromPVC(pvc)
-		if storage != "" {
-			size := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+		pvcComponentName, ok := pvc.Labels[componentlabels.ComponentLabel]
+		pvcAppName, okApp := pvc.Labels[applabels.ApplicationLabel]
+		if (!ok || pvcComponentName == componentName) && (okApp && pvcAppName == applicationName) {
+			if pvc.Name == "" {
+				return nil, fmt.Errorf("no PVC associated")
+			}
+			storageName := getStorageFromPVC(&pvc)
+			storageSize := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+			isMounted := false
```

Comment:
we don't needd `isMounted` flag, we can use`storagePath` as indicator whatever storage is mounted or not (if `storagePath` is empty or not)

Reply:
Ok then I will remove the isMounted flag